jenkins/igt/Jenkinsfile

```diff
@@ -0,0 +1,69 @@
+pipeline {
+  options { timestamps() }
+  agent { dockerfile { dir 'jenkins/igt/' } }
+
+  stages {
+    stage("Setup") {
+      steps {
+        script {
+          env.PIPELINE_VERSION = VersionNumber(
+             versionNumberString: '${BUILD_DATE_FORMATTED,"yyyyMMdd"}.${BUILDS_TODAY_Z}')
+        }
+        sh 'mkdir ${PIPELINE_VERSION}'
+      }
+    }
+
+    stage("Build libdrm") {
+      steps {
+        dir("${PIPELINE_VERSION}") {
+          sh 'mkdir libdrm'
+          dir("libdrm") {
+            git url: 'git://anongit.freedesktop.org/mesa/drm'
+            sh 'PKG_CONFIG_PATH=/usr/lib/arm-linux-gnueabihf/pkgconfig/ sh autogen.sh --host=arm-linux-gnueabihf --enable-intel --prefix=$(pwd)/../libdrm-build/'
+            sh 'make -j4 V=1'
+            sh 'make -j4 install V=1'
+            sh 'tar cvfz ../libdrm-build.tar.gz ../libdrm-build/'
+          }
+        }
+      }
+    }
+
+    stage("Build igt-gpu-tools") {
+      steps {
+        dir("${PIPELINE_VERSION}") {
+          sh 'mkdir igt-gpu-tools'
+          dir("igt-gpu-tools") {
+            git url: 'git://anongit.freedesktop.org/drm/igt-gpu-tools'
+            sh 'PKG_CONFIG_PATH=/usr/lib/arm-linux-gnueabihf/pkgconfig/:$(pwd)/../libdrm-build/lib/pkgconfig sh autogen.sh --host=arm-linux-gnueabihf'
+            sh 'make -j4 V=1'
+            sh 'tar cvfz ../igt-build.tar.gz tests/core_auth tests/core_get_client_auth tests/core_getclient tests/core_getstats tests/core_getversion tests/core_prop_blob tests/core_setmaster_vs_auth tests/drm_read tests/kms_addfb_basic tests/kms_atomic tests/kms_flip_event_leak tests/kms_setmode tests/kms_vblank tests/kms_frontbuffer_tracking tests/kms_flip tests/testdisplay'
+            sh ' rm -rf ../libdrm-build'
+            sh ' rm -rf ../libdrm'
+            deleteDir()
+          }
+        }
+      }
+    }
+
+    stage ("Upload image") {
+      environment {
+        NSS_WRAPPER_PASSWD = '/tmp/passwd'
+        NSS_WRAPPER_GROUP = '/dev/null'
+      }
+      steps {
+        sh 'ls -l'
+        script {
+          /* Only push if running as pipeline from SCM */
+          if (env.gitlabBranch == null) {
+            sh 'ln -sfn ${PIPELINE_VERSION} latest'
+            sshagent (credentials: [ "images.ccu-upload", ] ) {
+              sh 'echo docker:x:$(id -u):$(id -g):docker gecos:/tmp:/bin/false > ${NSS_WRAPPER_PASSWD}'
+              sh 'LD_PRELOAD=libnss_wrapper.so rsync -avz -e "ssh -o StrictHostKeyChecking=no" -r ${PIPELINE_VERSION} latest jenkins-upload@images.collabora.co.uk:/srv/jenkins-upload/image-uploads/lava/kci-igt-test'
```

Comment:
I think it may be better to use `storage.kernelci.org` rather than `images.collabora.co.uk`.  @mattface @khilman any thoughts on this?

Reply:
> From a hardware utilisation pov, how often would this build need to be run?
Hard to tell, to be honest, but I don't expect right now having to update the build more than a dozen times during this year.

> I think the impact of this single new build job should be small, but we don't have much headroom with our current builders. So as we add more builds for test suites and tools, we may need to source mode builder hardware to cope with the extra load.

In terms of load, I think this particular test suite should be really small compared with kernels. Shouldn't take much more than 5 mins. So close to noise when compared to the kernel builds.

But we'll be finding more regressions, so when we start triggering automatic bisections for test suites we'll require more builders because of this test plan.

Reply:
> I think it may be better to use storage.kernelci.org rather than images.collabora.co.uk. @mattface @khilman any thoughts on this?

I'm going to go ahead and upload them in a way similar to images in the next PR update.